### DISCUSSION TITLE
Build Dependencies for SEV-SNP VM: az-dcap

### DISCUSSION THREAD

**Discussion by Camelron** [microsoft/CCF#5756] [Q&A] [answered]
Hello,

I'm digging into the ansible scripts for installing dev dependencies and trying to refactor it for the OS image I am most familiar with (Mariner aka AzureLinux). 

Regarding the az-dcap component, https://github.com/microsoft/CCF/blob/main/getting_started/setup_vm/app-dev.yml#L19, do we need this for the SEV-SNP platform? If not, is there an analogous component for SEV-based enclaves?

**Comment by achamayou** [comment, marked as answer]
Hi @Camelron,

az-dcap-client is used to fetch SGX collateral when running in Azure, there is no analogous dependency for SEV-SNP, CCF reaches out to the Azure endpoints directly or gets it from env vars where available (ACI).

It's also not necessary to verify SGX quotes on SGX because CCF bundles collateral with attestation, so even in a mixed network (during a migration), it shouldn't be needed.

**Reply by Camelron** [reply]
Ah, I didn't realize you responded so quickly. Thanks!

**Comment by Camelron** [comment]
If you don't mind some follow-up questions:

How are the integrity of the 'instances' in the network verified when running on SEV-SNP systems? Do they need to be brought up in confidential VMs?

**Reply by achamayou** [reply]
@Camelron integrity is verified through attestation [at join time](https://microsoft.github.io/CCF/main/operations/start_network.html#adding-a-new-node-to-the-network), and so we require that it captures the [whole startup state](https://microsoft.github.io/CCF/main/operations/platforms/snp.html#amd-sev-snp) (VM image, client code running, [mounts if they exist etc](https://github.com/Azure/azure-cli-extensions/blob/main/src/confcom/azext_confcom/README.md#security-policy-rules-documentation)), which [Confidential Containers](https://learn.microsoft.com/en-us/azure/container-instances/container-instances-confidential-overview) do. We don't support other ways to run on SEV-SNP at the moment, but if they provided equivalent attestation coverage, we would certainly be interested.

Your questions are very welcome, we would like to explore running on Mariner once we've transitioned fully to SEV-SNP, but as we still need to support SGX for now we try to keep as much commonality between targets and we are stuck with Ubuntu 20.04 as a result.
